contracts/carbon/Strategies.sol

```diff
@@ -779,7 +781,7 @@ abstract contract Strategies is Initializable {
         uint256 temp4 = MathEx.mulDivC(temp1, temp1, factor);
         uint256 temp5 = MathEx.mulDivC(temp3, A, factor);
 
-        (bool safe, uint256 sum) = tryAdd(temp4, temp5);
+        (bool safe, uint256 sum) = SafeMathUpgradeable.tryAdd(temp4, temp5);
```

Comment:
Since you've declared `using SafeMathUpgradeable for uint256` above, you should do `temp4.tryAdd(temp5)`. BTW, this is probably why you've experienced a small increase in contract size, but it's ok, since this is how most of other libraries (e.g., `SafeCastUpgradeable`) are used. 

BTW, `MathUpgradeable` should be used this way as well

Reply:
`temp4.tryAdd(temp5)` is less clear than `tryAdd(temp4, temp5)` IMO.

Replaced all 3 "upgradable" libraries with their "non-upgradable" equivalence:
```
import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
import { SafeMath } from "@openzeppelin/contracts/utils/math/SafeMath.sol";
import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
```

Reply:
`temp4.tryAdd(temp5)` is how everything else is used, so this is the way to be consistent

Reply:
For the same reason, I've used `Math.max(factor1, factor2)` and not `factor1.max(factor2)`.

Same goes for using `MathEx.mulDivC(x, y, z)` instead of `x.mulDivC(y, `)`, etc.

Reply:
Remember using `SafeMath` for `add`, `sub`, `mul`, etc.? It was always `x.add(y)` and not `add(x, y)`. In any case, let's let @yudilevi to decide on this one

Reply:
Both are fine, let’s go with the cheaper version

Reply:
@barakman make your choice and I'll quickly review and approve ❤️

Reply:
I have previously removed a couple of `using` statement which were not used in the contract:
- `using Math`
- `using SafeMath`

While leaving in place the `using` statement which WAS used in the contract:
- `using SafeCast`

If that's what you meant, then it's already done.

Reply:
Yeah, this is the first approach